micrometer-tracing-reporters/micrometer-tracing-reporter-wavefront/src/main/java/io/micrometer/tracing/reporter/wavefront/SpanMetrics.java

```diff
@@ -17,18 +17,71 @@
 package io.micrometer.tracing.reporter.wavefront;
 
 import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.atomic.AtomicLong;
 
 /**
+ * Reports metrics from {@link WavefrontSpanHandler}.
+ *
  * @author Moritz Halbritter
+ * @since 1.0.0
  */
 public interface SpanMetrics {
-    long reportDropped();
+	/**
+	 * Is called when a span has been dropped.
+	 *
+	 * @return number of total dropped spans
+	 */
+	long reportDropped();
+
+	/**
+	 * Is called when a span is received.
+	 */
+	void reportReceived();
+
+	/**
+	 * Is called when a span couldn't be sent.
+	 */
+	void reportErrors();
+
+	/**
+	 * Registers the size of the given {@code queue}.
+	 *
+	 * @param queue queue which size should be registered
+	 */
+	void registerQueueSize(BlockingQueue<?> queue);
+
+	/**
+	 * Registers the remaining capacity of the given {@code queue}.
+	 *
+	 * @param queue queue which remaining capacity should be registered
+	 */
+	void registerQueueRemainingCapacity(BlockingQueue<?> queue);
+
+	/**
+	 * No-op implementation.
+	 */
+	SpanMetrics NOOP = new SpanMetrics() {
+		private final AtomicLong dropped = new AtomicLong();
+
+		@Override
+		public long reportDropped() {
+			return dropped.incrementAndGet();
```

Comment:
This is noot noop, is this used or can this be `return 0;`?

Reply:
Well, in theory you could, but then the [log messages will be wrong](https://github.com/micrometer-metrics/tracing/pull/12/files/dc8154f5821dbe082cb373268a187e2b94189f4e#diff-744d1e99ee06f63da5f610c0c1f539bd1f1aa2321e2f05e8ecd64855865be900R272). I could remove the counter from the `SpanMetrics` and put them into `WavefrontSpanHandler` directly. This would make the `SpanMetrics` only react to events and it has no internal state anymore, making the no-op implementation a real no-op. Should I do this?

Reply:
I've moved the counter for the log message into `WavefrontSpanHandler`. I think this is cleaner, because now the log message can't be wrong when a `SpanMetrics` implementation has a bug. This makes the no-op implementation a real noop.